File: handlers/page_load_login.zig
const std = @import("std");
const t = @import("../prelude.zig");

pub const Status = enum { ok };

pub const Prefetch = struct {};

pub const Context = t.HandlerContext(Prefetch, t.Operation.EventType(.page_load_login), t.Identity, Status);

// [route] .page_load_login
// match GET /login
pub fn route(params: t.RouteParams, body: []const u8) ?t.Message {
    _ = params; _ = body;
    return t.Message.init(.page_load_login, 0, 0, {});
}

// [prefetch] .page_load_login
pub fn prefetch(storage: anytype, msg: *const t.Message) ?Prefetch {
    _ = storage; _ = msg;
    return .{};
}

// [handle] .page_load_login
pub fn handle(ctx: Context, db: anytype) t.HandleResult {
    _ = ctx;
    _ = db;
    return .{};
}


// [render] .page_load_login
pub fn render(ctx: Context) []const u8 {
    _ = ctx;
    return "<form id=\"login-box\"><input name=\"email\" placeholder=\"Email\"/><button>Send Login Code</button></form>";
}

File: examples/ecommerce-ts/handlers/page_load_login.ts
import type { RouteRequest, RouteResult, PrefetchMessage, PrefetchDb, HandleContext, RenderContext } from "focus";

// [route] .page_load_login
// match GET /login
export function route(_req: RouteRequest): RouteResult | null {
  return { operation: "page_load_login", id: "0".repeat(32) };
}

// [prefetch] .page_load_login
export async function prefetch(_msg: PrefetchMessage, _db: PrefetchDb) {
  return {};
}

// [handle] .page_load_login
export function handle(_ctx: HandleContext): string {
  return "ok";
}

// [render] .page_load_login
export function render(_ctx: RenderContext): string {
  return "<div>Login</div>";
}
